kloppy/infra/serializers/event/metrica/json_serializer.py

```diff
@@ -0,0 +1,311 @@
+        result=CarryResult.COMPLETE,
+        end_coordinates=_parse_coordinates(event["end"]),
+        end_timestamp=event["end"]["time"],
+    )
+
+
+def _parse_take_on(subtypes: List) -> Dict:
+    if MS_WON in subtypes:
+        result = TakeOnResult.COMPLETE
+    else:
+        result = TakeOnResult.INCOMPLETE
+
+    return dict(result=result)
+
+
+def _parse_ball_owning_team(event_type: int, team: Team) -> Team:
+    if event_type not in [
+        MS_EVENT_TYPE_CHALLENGE,
+        MS_EVENT_TYPE_CARD,
+    ]:
+        return team
+    else:
+        return None
+
+
+class MetricaEventsJsonSerializer(EventDataSerializer):
+    @staticmethod
+    def __validate_inputs(inputs: Dict[str, Readable]):
+        if "event_data" not in inputs:
+            raise ValueError("Please specify a value for input 'event_data'")
+        if "metadata" not in inputs:
+            raise ValueError("Please specify a value for input 'metadata'")
+
+    def deserialize(
+        self, inputs: Dict[str, Readable], options: Dict = None
+    ) -> EventDataset:
+        """
+                Deserialize Metrica Sports event data json format into a `EventDataset`.
+
+                Parameters
+                ----------
+                inputs : dict
+                    input `event_data` should point to a `Readable` object containing
+                    the 'json' formatted event data. input `metadata` should point
+                    to a `Readable` object containing the `xml` metadata file.
+                options : dict
+                    Options for deserialization of the Metrica Sports event json file. 
+                    Possible options are `event_types` (list of event types) to specify 
+                    the event types that should be returned. Valid types: "shot", "pass", 
+                    "carry", "take_on" and "generic". Generic is everything other than 
+                    the first 4. Those events are barely parsed. This type of event can 
+                    be used to do the parsing yourself.
+                    Every event has a 'raw_event' attribute which contains the original
+                    dictionary.
+                Returns
+                -------
+                dataset : EventDataset
+                Raises
+                ------
+
+                See Also
+                --------
+
+                Examples
+                --------
+                >>> serializer = MetricaEventsJsonSerializer()
+                >>> with open("events.json", "rb") as event_data, \
+                >>>      open("metadata.xml", "rb") as metadata:
+                >>>
+                >>>     dataset = serializer.deserialize(
+                >>>         inputs={
+                >>>             'event_data': event_data,
+                >>>             'metadata': metadata
+                >>>         },
+                >>>         options={
+                >>>             'event_types': ["pass", "take_on", "carry", "shot"]
```

Comment:
Outside the scope of this PR, and looks like it's how it's done for the opta and statsbomb serializers currently, but is there a reason why this shouldn't just be a list of EventType enums?

Reply:
Yes. Better to pass the EventTypes here and do the resolve from string to EventType in the helper function. I don’t want to bother users with the EventTypes because that takes additional imports.